src/reducers/vis-state-merger.js

```diff
@@ -52,43 +51,49 @@ export function mergeFilters(state, filtersToMerge) {
 
   // merge filters
   filtersToMerge.forEach(filter => {
-    // match filter.dataId with current datesets id
-    // uploaded data need to have the same dataId with the filter
-    if (datasets[filter.dataId]) {
-      // datasets is already loaded
-      const validateFilter = validateFilterWithData(
-        datasets[filter.dataId],
-        filter
-      );
+    const datasetIds = Object.keys(datasets).reduce((acc, datasetId) => ([
```

Comment:
why use reduce when this is just a simple filter function?
`datasetIds = Object.keys(datasets).filter(datasetId => shouldApplyfilter(filter, datasetId))`